src/App.js

```diff
@@ -26,22 +55,39 @@ class App extends Component {
             systemDetailReducer,
             systemRecsReducer });
         insights.chrome.init();
-        // TODO change this to your appname
         insights.chrome.identifyApp('ros');
-
         this.appNav = insights.chrome.on('APP_NAVIGATION', event => this.props.history.push(`/${event.navId}`));
+
+        (async () => {
+            const rosPermissions = await insights.chrome.getUserPermissions('ros');
+            const permissionsList = rosPermissions.map(permissions => permissions.permission);
```

Comment:
Extra roundtrip. On line 65 you can do directly `permissionsList.some(({ permission }) => this.hasPermission(permission, ['ros:*:*', 'ros:*:read']))`